gradle-plugins/plugin-base/src/test/java/io/spine/gradle/SpinePluginBuilderShould.java

```diff
@@ -0,0 +1,151 @@
+        final Collection<?> dependencies = newTask.getDependsOn();
+        assertTrue(dependencies.contains(subProjectTasks.findByName(COMPILE_JAVA.getValue())));
+        assertTrue(dependencies.contains(project.getTasks()
+                                                .findByName(COMPILE_JAVA.getValue())));
+    }
+
+    @Test
+    public void create_task_and_insert_before_other() {
+        final SpinePlugin plugin = TestPlugin.INSTANCE;
+        plugin.newTask(VERIFY_MODEL, NoOp.<Task>action())
+              .insertBeforeTask(CLASSES)
+              .applyNowTo(project);
+        final TaskContainer tasks = project.getTasks();
+        final Task classes = tasks.findByName(CLASSES.getValue());
+        final Task verifyModel = tasks.findByName(VERIFY_MODEL.getValue());
+        assertTrue(classes.getDependsOn().contains(verifyModel));
+    }
+
+    @Test
+    public void create_task_and_insert_after_other() {
+        final SpinePlugin plugin = TestPlugin.INSTANCE;
+        plugin.newTask(VERIFY_MODEL, NoOp.<Task>action())
+              .insertAfterTask(COMPILE_JAVA)
+              .applyNowTo(project);
+        final TaskContainer tasks = project.getTasks();
+        final Task compileJava = tasks.findByName(COMPILE_JAVA.getValue());
+        final Task verifyModel = tasks.findByName(VERIFY_MODEL.getValue());
+        assertTrue(verifyModel.getDependsOn().contains(compileJava.getName()));
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void not_allow_tasks_without_any_connection_to_task_graph() {
+        TestPlugin.INSTANCE.newTask(FIND_VALIDATION_RULES, NoOp.<Task>action()).applyNowTo(project);
+    }
+
+    @Test
+    public void return_build_task_description() {
+        final SpinePlugin plugin = TestPlugin.INSTANCE;
+        final SpinePlugin.GradleTask desc = plugin.newTask(PRE_CLEAN, NoOp.<Task>action())
+                                                  .insertBeforeTask(CLEAN)
+                                                  .applyNowTo(project);
+        assertEquals(PRE_CLEAN, desc.getName());
+        assertEquals(project, desc.getProject());
+    }
+
+    @Test
+    public void create_task_with_given_inputs() throws IOException {
+        final SpinePlugin plugin = TestPlugin.INSTANCE;
+        final File input = new File(".").getAbsoluteFile();
+        plugin.newTask(PRE_CLEAN, NoOp.<Task>action())
+              .insertBeforeTask(CLEAN)
+              .withInputFiles(input.toPath())
+              .applyNowTo(project);
+        final Task task = project.getTasks().findByPath(PRE_CLEAN.getValue());
+        final File singleInput = task.getInputs()
+                                      .getFiles()
+                                      .getFiles()
+                                      .iterator()
+                                      .next();
+        assertEquals(input.getCanonicalFile(), singleInput.getCanonicalFile());
+    }
+
+    private static class TestPlugin extends SpinePlugin {
```

Comment:
With this name this class should at least have a class-level comment on the purpose of creation.